storm-client/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -74,6 +74,13 @@
         return gauge;
     }
 
+    public Meter meter(String name, String topologyId, String componentId, Integer taskId, Integer workerPort, String streamId) {
```

Comment:
Maybe we should remove `streamId` every where since it is inside the metric name. 
Or we explicitly combine metricName+srcComponent+streamId as the new metric Name in `workerMetricName` method to make it consistently (ignore when the certain variable is `null`)

Reply:
I am not following what you are indicating.  If I remove the streamId, how could it be part of the metric name?

Reply:
I thought you added streamId to the name, for example, `__acked-count-comp1:stream1`
 
So I was thinking to  make it consistent, move that logic to a more general function like here. Instead of changing the name before calling this function.